en/sensor/pmw3901.md

```diff
@@ -0,0 +1,119 @@
+GPS was fully disabled in this test to see the real performance of the flow module.
+
+**REPLACE WITH RIGHT VIDEO!**
+{% youtube %}
+{% endyoutube %}
+
+## Boards using PMW3901
+
+The following table shows some of the boards that use this sensor, listing the number of interfaces, the number of sensors, the input voltage, and the size.
+The board name contains the link to where to buy the specific board.
+
+Manufacture | Board | Interface | Flow | Range Finder | Gyro | Voltage (V) | Size (mm) | Max Height (m)
+--- | --- | --- | --- | --- | --- | --- | ---
+<span id="Bitcraze"></span>Bitcraze | [Flow breakout](https://www.bitcraze.io/flow-breakout) | SPI | X | X | - | 3 - 5 | 21x20 | 1
+<span id="Tindie"></span>Tindie | [PMW3901 Optical Flow Sensor](https://www.tindie.com/products/onehorse/pmw3901-optical-flow-sensor/) | SPI | X | - | - | 3 - 5 | AxB | -
+<span id="Hex"></span>Hex | [HereFlowPMW3901 Optical Flow Sensor](http://www.proficnc.com/all-products/185-pixhawk2-suite.html) | CAN | X | X | X | 3 - 5 | AxB | 4
+<span id="Thone"></span>Thone | [ThoneFlow-3901U](https://www.seeedstudio.com/ThoneFlow-3901U-UART-Serial-Version-PMW3901-Optical-Flow-Sensor-p-4040.html) | UART | X | - | - | 3 - 5 | AxB | -
+<span id="Alientek"></span>Alientek | [ATK-PMW3901](https://www.aliexpress.com/i/32979605707.html) | SPI | X | - | - | 3.3 - 4.2 | 27.5x16.5 | -
+
+An external [rangefinder/distance sensor](../sensor/rangefinders.md) is *required* for the boards that don't have a rangefinder (e.g. *Tindie* or *Thone*) and recommended for the other boards (as their range is quite limited).
+
+## Specifications
+
+* TODO
+* TODO
+* TODO
+
+## Pixhawk Setup
+
+### Bitcraze Flow breakout {#bitcraze}
+
+The Bitcraze optical flow module directly exposes the SPI interface from the PMW3901 module.
+
+The board also has a distance sensor that can be wired via I2C.
+This distance sensor is the VL53L0x ToF sensor from STMicroelectronics.
+The range of the sensor is very limited (2 meters) and will be reduced when flying in the sunlight.
+We therefore highly recommend to use another external distance sensor.
+This board was mainly tested with the [Lidar Lite V3](../sensor/lidar_lite.md) on larger vehicles and with the [Lanbao CM8JL65](../sensor/cm8jl65_ir_distance_sensor.md) on smaller vehicles.
+
+The range needed is dependent on the application.
+When flying indoors, generally 4 meters height are enough.
+If optical flow is also needed outdoors to support the position control in environments where GPS might have issues, then more range is advised (10 meters or more).
+
+The PMW3901 driver was written to be directly plugged into the SPI port on the Pixhawk 4 using the chip select 1.
+The support for this driver on other boards can then easily be extended to other boards by modifying the `SPI_BUS` and the `SPI_DEV` in the [PMW3901 driver](https://github.com/PX4/Firmware/tree/{{ book.px4_version}}/src/drivers/optical_flow/pmw3901).
```

Comment:
Just because it was never tested on anything different from a Pixhawk 4 or a Pixracer. I can add the wiring for the pixracer. SPI is a bit of a pain because we have different busses and different bus names (the defines in the board_config.h) therefore we would actually need to try it on all boards.